File: app/src/main/java/net/radityalabs/contactapp/presentation/rx/transformation/Transformation.java
package net.radityalabs.contactapp.presentation.rx.transformation;

import io.reactivex.FlowableTransformer;

/**
 * Created by radityagumay on 4/12/17.
 */

public interface Transformation {

    <T> FlowableTransformer<T, T> loading();

    <T> FlowableTransformer<T, T> loading(String message);
}

File: app/src/main/java/net/radityalabs/contactapp/presentation/rx/transformation/Transformation.kt
package net.radityalabs.contactapp.presentation.rx.transformation

import io.reactivex.FlowableTransformer

/**
 * Created by radityagumay on 4/12/17.
 */

interface Transformation {

    fun <T> loading(): FlowableTransformer<T, T>

    fun <T> loading(message: String): FlowableTransformer<T, T>
}
